jinja2/filters.py

```diff
@@ -518,9 +518,17 @@ def do_int(value, default=0, base=10):
     can also override the default base (10) in the second
     parameter, which handles input with prefixes such as
     0b, 0o and 0x for bases 2, 8 and 16 respectively.
+    The base is ignored for non-string values.
     """
+    # only string types support int() with explicit base
+    if isinstance(value, string_types) and base != 10:
+        try:
+            return int(value, base)
+        except ValueError:
+            return default
```

Comment:
I kind of understand that rationale but changing the API is inappropriate for a bugfix, so I'd like to keep the previous behavior (`{{ "42.3" | int(0, 16) }}` is `42`). Feel free to open an issue about this though.